doc/ui_component2_transition.md

```diff
@@ -0,0 +1,135 @@
+
+### Lifecycle Method Changes
+
+- Drops support for unsafe lifecycle methods:
+  - componentWillMount
+  - componentWillReceiveProps
+  - componentWillUpdate
+- Adds support for new, safe lifecycle methods:
+  - getDerivedStateFromProps
+  - getSnapshotBeforeUpdate
+  - componentDidCatch / getDerivedStateFromError (adds support for [error boundaries](https://reactjs.org/docs/error-boundaries.html))
+- componentDidUpdate now receives a `snapshot` argument, which is the return value of getSnapshotBeforeUpdate.
+  
+See the [React docs](https://reactjs.org/docs/react-component.html#the-component-lifecycle) for information on why the old methods are unsafe and how to transition to the new ones. This 
+information is also available and linked to within the lifecycle methods' doc comments.
+
+### Other Method Changes
+- `getDefaultProps` is now `get defaultProps`
+- `getInitialState` is now `get initialState`
+- `..addProps(copyUnconsumedProps())` is now `..modifyProps(addUnconsumedProps)`. The same is true with `DomProps`, 
+the usage being `..modifyProps(addUnconsumedDomProps)`. Using `addUnconsumedProps` was shown to render 8% faster than
+ `copyUncosumedProps`.
+
+### Prop validation via PropTypes
+- Aligns with ReactJS implementation.
+- Shows "component stack" that helps determine where the error came from
+- React automatically de-duplicates the same warnings
+- __No longer throws, but warns__. When upgrading to `Component2`, prop validation will no longer stop the component 
+from rendering. If tests exist verifying that prop validation occurs, these will likely need to change. There are new 
+`PropType` testing utils within [over_react_test](https://github.com/Workiva/over_react_test).
+
+### ErrorBoundary Component Added
+
+With the ability to utilize componentDidCatch / getDerivedStateFromError comes the ability to use [error boundaries](https://reactjs.org/docs/error-boundaries.html).
+A component (called ErrorBoundary) is present within OverReact and can be used to wrap a component tree to prevent it 
+from unmounting when an error is thrown. Alternatively, a custom error boundary can be created by utilizing the 
+[ErrorBoundaryMixin](../lib/src/component/error_boundary_mixins.dart).
+
+Usage of the default error boundary can be found in the Component1 examples (within web), and usage of a custom error 
+boundary can be found in the Component2 examples.
+
+### Other breakages we made while we had the opportunity
+- Split `Component.setState` into two methods, so that each has strong typing 
+  - setState's first argument is now typed as `Map`, not `dynamic`
+  - A new `setStateWithUpdater` is added, which accepts transactional updater functions
+- Remove `bind`—a non-standard API leftover from early react-dart code—to reduce perf overhead
+- Re-implement prop validation and remove support for `validateProps`
+- Drop support for legacy context APIs
+
+## Updating
+
+To update your repository to use UiComponent2 / Component2, you can use 
```

Comment:
I think it'd be worth making it clear up front that the codemod won't automatically upgrade everything perfectly.